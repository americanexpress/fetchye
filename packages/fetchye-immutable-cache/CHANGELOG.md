# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-beta.5](https://github.com/americanexpress/fetchye/compare/v1.0.0-beta.3...v1.0.0-beta.5) (2020-11-04)



# 1.0.0-beta.4 (2020-11-03)

**Note:** Version bump only for package fetchye-immutable-cache





# [1.0.0-beta.4](https://github.com/americanexpress/fetchye/compare/v1.0.0-beta.3...v1.0.0-beta.4) (2020-11-03)

**Note:** Version bump only for package fetchye-immutable-cache
