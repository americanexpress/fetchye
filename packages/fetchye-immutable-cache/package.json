{
  "name": "fetchye-immutable-cache",
  "version": "1.0.3",
  "description": "Immutable cache library to use with fetchye.",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "module": "es/index.js",
  "files": [
    "lib",
    "es"
  ],
  "scripts": {
    "clean": "rimraf lib && rimraf es",
    "prebuild": "yarn clean",
    "build:cjs": "cross-env BABEL_ENV=commonjs babel --extensions .js,.jsx ./src --out-dir lib",
    "build:esm": "babel --extensions .js,.jsx ./src --out-dir es",
    "build": "yarn build:cjs && yarn build:esm"
  },
  "keywords": [
    "fetchye-immutable-cache",
    "immutable",
    "fetchye"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/americanexpress/fetchye.git",
    "directory": "packages/fetchye-immutable-cache"
  },
  "contributors": [
    "James Singleton <James.Singleton1@aexp.com> https://github.com/JamesSingleton)",
    "Jimmy King  <Jimmy.King@aexp.com> (https://github.com/10xLaCroixDrinker)",
    "Jonathan Adshead <Jonathan.Adshead@aexp.com> (https://github.com/JAdshead)",
    "Michael Tobia <Michael.M.Tobia@aexp.com> (https://github.com/Francois-Esquire)",
    "Nelly Kiboi <Nelly.J.Kiboi@aexp.com> (https://github.com/nellyk)",
    "Ruben Casas <ruben.casas@aexp.com> (https://github.com/infoxicator)"
  ],
  "dependencies": {
    "immutable": "^4.0.0-rc.12"
  },
  "devDependencies": {
    "@babel/cli": "^7.12.1",
    "@babel/core": "7.11.6",
    "babel-preset-amex": "^3.4.1",
    "cross-env": "^7.0.2",
    "fetchye-test-utils": "^1.0.3",
    "redux": "^4.0.5",
    "rimraf": "^3.0.2"
  },
  "peerDependencies": {
    "fetchye": "^1.0.3",
    "fetchye-core": "^1.0.3"
  }
}
