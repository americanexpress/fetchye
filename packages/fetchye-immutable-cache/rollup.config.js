/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { nodeResolve } from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import babel from '@rollup/plugin-babel';
import { terser } from 'rollup-plugin-terser';
import pkg from './package.json';

const inputSrc = 'src/index.js';

export default [
  {
    input: inputSrc,
    output: {
      name: 'fetchye-immutable-cache',
      globals: {
        react: 'React',
      },
      file: pkg.browser,
      format: 'umd',
    },
    plugins: [
      babel({
        babelHelpers: 'bundled',
        exclude: 'node_modules/**',
      }),
      nodeResolve({ browser: true }),
      commonjs(),
      terser(),
    ],
  },
  {
    input: inputSrc,
    output: {
      file: pkg.main,
      format: 'cjs',
    },
    plugins: [
      babel({
        babelHelpers: 'bundled',
        exclude: 'node_modules/**',
      }),
      nodeResolve(),
      terser(),
    ],
  },
  {
    input: inputSrc,
    output: {
      file: pkg.module,
      format: 'es',
    },
    plugins: [
      babel({
        babelHelpers: 'bundled',
        exclude: 'node_modules/**',
      }),
      nodeResolve(),
      terser(),
    ],
  },
];
