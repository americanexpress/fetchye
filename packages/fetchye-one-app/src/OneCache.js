/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { ImmutableCache } from 'fetchye-immutable-cache';
import invariant from 'invariant';

export const oneCacheSelector = (state) => {
  const rootModuleName = state.getIn(['config', 'rootModuleName']);
  return state.getIn(['modules', rootModuleName, 'fetchye']);
};

export default function OneCache(args = {}) {
  invariant(
    !args.cacheSelector,
    '"cache" is not a permitted argument for OneCache, please use the ImmutableCache provided by fetchye-immutable-cache: https://github.com/americanexpress/fetchye#immutablecache'
  );
  return ImmutableCache({
    cacheSelector: oneCacheSelector,
  });
}
