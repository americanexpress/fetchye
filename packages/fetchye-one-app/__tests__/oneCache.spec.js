/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { fromJS } from 'immutable';
import { ImmutableCache } from 'fetchye-immutable-cache';
import OneCache, { oneCacheSelector } from '../src/OneCache';

jest.mock('fetchye-immutable-cache', () => ({
  ImmutableCache: jest.fn(),
}));

describe('oneCache', () => {
  it('uses ImmutableCache', () => {
    OneCache();
    expect(ImmutableCache).toHaveBeenCalledWith({
      cacheSelector: oneCacheSelector,
    });
  });

  it('throws when cacheSelector provided', () => {
    expect(() => OneCache({ cacheSelector: 'some-selector' })).toThrowErrorMatchingSnapshot();
  });

  describe('oneCacheSelector', () => {
    it('uses root module name to select state', () => {
      const rootModuleName = 'some-root-modules-name';
      const state = fromJS({
        config: { rootModuleName },
        modules: {
          [rootModuleName]: { fetchye: 'fetchye-cache' },
        },
      });
      const fetchyeState = oneCacheSelector(state);
      expect(fetchyeState).toEqual('fetchye-cache');
    });
  });
});
