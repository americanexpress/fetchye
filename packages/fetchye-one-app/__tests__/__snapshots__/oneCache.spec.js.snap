// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`oneCache throws when cacheSelector provided 1`] = `"\\"cache\\" is not a permitted argument for OneCache, please use the ImmutableCache provided by fetchye-immutable-cache: https://github.com/americanexpress/fetchye#immutablecache"`;
